### PR TITLE
Python-preparation

### PR DESCRIPTION
**Pull request by jdroenner**

### CODE REVIEW

**Comment by jdroenner** [comment]
[Python Bugfixes + repr](https://trello.com/c/5jbKMrUO/1209-python-bugfixes-repr)

File: geoengine/datasets.py [hunk 1]
```diff
@@ -266,8 +266,8 @@ def from_response(cls, response: api.DatasetId) -> DatasetId:
         if 'error' in response:
             raise GeoEngineException(cast(api.GeoEngineExceptionResponse, response))
 
-        if 'id' not in response:  # TODO: improve error handling
-            raise Exception('No id in response')
+        if 'id' not in response:
+            raise UnexpectedResponseException('No id in response', response)
```
Comment (michaelmattig) [inline geoengine/datasets.py:270]: maybe a MissingFieldInResponseExcpetion would be better?

File: geoengine/error.py [hunk 2]
```diff
@@ -137,6 +137,38 @@ def __str__(self) -> str:
         return f"Spatial reference mismatch {self.__spatial_reference_a} != {self.__spatial_reference_b}"
 
 
+class NoValidUrlException(Exception):
```
Comment (michaelmattig) [inline geoengine/error.py:140]: maybe InvalidUrlException would be better

File: geoengine/datasets.py (hunk 1 above)
Comment (jdroenner) [reply geoengine/datasets.py]: [d825fa0](https://github.com/geo-engine/geoengine-python/pull/102/commits/d825fa04cb26a20bebd9fbab2e1aa3d54d18af97)

File: geoengine/error.py (hunk 2 above)
Comment (jdroenner) [reply geoengine/error.py]: [d825fa0](https://github.com/geo-engine/geoengine-python/pull/102/commits/d825fa04cb26a20bebd9fbab2e1aa3d54d18af97)

File: geoengine/datasets.py (hunk 1 above)
Comment (michaelmattig) [reply geoengine/datasets.py]: I had a typo in there "Excpetion" which you copied. Sorry about that.

Comment (jdroenner) [reply geoengine/datasets.py]: [26fad8d](https://github.com/geo-engine/geoengine-python/pull/102/commits/26fad8dbd3c6f7319694f93d089d6cc43800e802)

**Review by michaelmattig — APPROVED** [review]

**Comment by michaelmattig** [comment]
bors merge

**Pull request merged**